bitcoin/src/taproot/merkle_branch/borrowed.rs

```diff
@@ -0,0 +1,290 @@
+use core::borrow::{Borrow, BorrowMut};
+use internals::slice::SliceExt;
+
+use super::{DecodeError, InvalidMerkleBranchSizeError, InvalidMerkleTreeDepthError, TaprootMerkleBranchBuf, TapNodeHash, TAPROOT_CONTROL_MAX_NODE_COUNT, TAPROOT_CONTROL_NODE_SIZE};
+
+pub use privacy_boundary::TaprootMerkleBranch;
+
+/// Makes sure only the allowed conversions are accessible to external code.
+mod privacy_boundary {
```

Comment:
I don't see the point of this module.

Reply:
It prevents direct access to the `.0` field. Though the comment is somewhat misleading because any "disallowed" conversion would have to use `unsafe` anyway.